### PR TITLE
I did some stuff

### LINKED ISSUE #6: Make Database Integration Private
Currently the middleware and service handlers could have access to the database because functions like `look_up_user` are public. They should only be callable when accessed through the `Backend` trait implementation.

### LINKED ISSUE #7: Doctests in Workspace Crates Are Not Run

### PR DESCRIPTION
**Pull request by VoigtSebastian**

### CODE REVIEW

**Comment by VoigtSebastian** [comment]
I will fix this later

**Comment by VoigtSebastian** [comment]
I installed [rust-hook](https://crates.io/crates/rusty-hook) which seems like an interesting crate.

**Comment by VoigtSebastian** [comment]
This pr will close #6 and #7 when merged

**Comment by VoigtSebastian** [comment]
Turns out, if you want to upgrade sqlx to version 0.5 you will have to upgrade actix-web and actix-rt.
This breaks the middleware implementation and the current documentation of actix-web 4 (beta) isn't great.

File: .github/workflows/test-all.yml [hunk 1]
```diff
@@ -45,30 +53,39 @@ jobs:
...
+      - name: Run cargo clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --workspace -- -D warnings
+      - name: Run cargo doc test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --workspace --doc
```
Comment (b-faller) [inline .github/workflows/test-all.yml]: The doc argument can be dropped. See the comment in #7.

File: automation.sh [hunk 2]
```diff
@@ -115,6 +115,9 @@ elif [ "$1" == "psql-uri" ]; then
 elif [ "$1" == "test" ]; then
     cargo test --workspace -- --ignored
 
+elif [ "$1" == "doctest" ]; then
+    cargo test --workspace --doc
```
Comment (b-faller) [inline automation.sh]: We could use `cargo test --workspace -- --include-ignored` (#7).

**Review by b-faller — CHANGES_REQUESTED** [review]
Looks good so far.

**Review by b-faller — APPROVED** [review]

**Pull request merged**